pallets/apollo-platform/src/lib.rs

```diff
@@ -1565,6 +1642,33 @@ pub mod pallet {
             Ok(().into())
         }
 
+        fn calculate_max_allowed_collateral(
+            lending_amount: Balance,
+            factor: Balance,
+        ) -> Result<Balance, DispatchError> {
+            Ok(
+                (FixedWrapper::from(lending_amount) * FixedWrapper::from(factor))
+                    .try_into_balance()
+                    .unwrap_or(0),
+            )
+        }
+
+        /// Calculate the total existing collateral for a specific user and asset
+        fn calculate_total_existing_collateral(
+            user: &AccountIdOf<T>,
+            collateral_asset: &AssetIdOf<T>,
+        ) -> Balance {
+            <UserBorrowingInfo<T>>::iter()
```

Comment:
I think it's better to aggregate total collateral amount in separate storage

Reply:
Hello,
Thanks for the suggestion, we will do it, but it will take some time since we need to write migrations as well, to migrate the state of current users to the new storage that we create for it.